### PR TITLE
feat(schemas): schema name prefix flag and remote connection behavior

### PR DESCRIPTION
**Pull request by vitaligi**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

- add a flag to control schema prefix
- change the behavior of remote connection only when at least "enableExternalFetch": "yes"

### CODE REVIEW

File: src/containerConfig.ts [hunk 1]
```diff
@@ -24,25 +24,35 @@ async function registerExternalValues(): Promise<void> {
...
+    redisConnection = createConnection(config.get<RedisOptions>('db.connection.options'));
+    redisConnection.on('error', (err) => {
+      logger.fatal(err, `Redis connection failure, exiting with code ${REDIS_CONNECTION_ERROR_CODE}`);
+      return process.exit(REDIS_CONNECTION_ERROR_CODE);
+    });
+    container.register(REDIS_SYMBOL, { useValue: redisConnection });
+    container.register(IDOMAIN_FIELDS_REPO_SYMBOL, { useClass: RedisManager });
+  } else {
+    container.register(IDOMAIN_FIELDS_REPO_SYMBOL, { useValue: {} });
+  }
```
Comment (CptSchnitz) [inline src/containerConfig.ts:47]: I don't think redis connection should depend on any schema using it. If we want the option to turn it off, it should be a configuration option. WDYT @syncush ?

File: src/schema/models/schemaManager.ts [hunk 2]
```diff
@@ -81,42 +88,49 @@ export class SchemaManager {
...
+  private readonly getDomainFields = async (redisKeysArr: string[], name: string, addSchemaPrefix: boolean): Promise<Tags> => {
     let domainFieldsTags: Tags = {};
     const redisRes = await this.domainFieldsRepo.getFields(redisKeysArr);
 
     //for each domain field create new domain field tag with the correct value
     redisRes.forEach((key, index) => {
-      domainFieldsTags = { ...domainFieldsTags, [`${name}_${redisKeysArr[index].split(':')[0]}_DOMAIN`]: key };
+      domainFieldsTags = {
+        ...domainFieldsTags,
+        [`${this.concatenateKeysPrefix(addSchemaPrefix, name, redisKeysArr[index].split(':')[0], 'DOMAIN')}`]: key,
```
Comment (CptSchnitz) [inline src/schema/models/schemaManager.ts:119]: There is no need to template the string.

Comment (vitaligi) [reply src/schema/models/schemaManager.ts:119]: done

**Review by syncush — CHANGES_REQUESTED** [review]

**Review by syncush — CHANGES_REQUESTED** [review]
You are still invoking the prefix concatenation many times instead of once.

**Pull request merged**